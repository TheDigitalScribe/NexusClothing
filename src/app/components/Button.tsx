import React from 'react'

interface Button {
  text: string;
  ariaLabel: string;
}

export const Button = ({ text, ariaLabel }: Button) => {
  return (
    <button aria-label={ariaLabel} className="rounded-full py-2 px-4 bg-blue-600 items-center text-white shadow-sm transition-colors duration-300">{text}</button>
  )
}
