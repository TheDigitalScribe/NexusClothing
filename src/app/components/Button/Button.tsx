import React from 'react'

interface Button {
  text: string;
  ariaLabel: string;
  onClick: () => void;
}

export const Button = ({ text, ariaLabel, onClick }: Button) => {
  return (
    <button onClick={onClick} aria-label={ariaLabel} className="rounded-full py-2 px-4 bg-blue-600 items-center text-white shadow-sm transition-colors duration-300">{text}</button>
  )
}
