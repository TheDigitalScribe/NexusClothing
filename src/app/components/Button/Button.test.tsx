import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Button } from './Button';

const mockProps = {
  text: 'Test Text',
  ariaLabel: 'Test Aria Label',
}

describe('Button', () => {
  // Render test
  it('renders the component', () => {
    render(<Button {...mockProps} />);

    expect(screen.getByText('Test Text')).toBeInTheDocument();
  });

  // Tailwind CSS classes test
  it('applies correct Tailwind CSS classes', () => {
    render(<Button {...mockProps} />);

    expect(screen.getByText(mockProps.text)).toHaveClass('rounded-full py-2 px-4 bg-blue-600 items-center text-white shadow-sm transition-colors duration-300');
  });

  // Correct aria-label test
  it('has correct aria-label', () => {
    render(<Button {...mockProps} />);

    const button = screen.getByRole('button', { name: mockProps.ariaLabel });
    expect(button).toHaveAttribute('aria-label', mockProps.ariaLabel);
  });

  // Responsive classes test
  it('has correct responsive classes', () => {
    render(<Button {...mockProps} />);

    expect(screen.getByText(mockProps.text)).toHaveClass('rounded-full py-2 px-4 bg-blue-600 items-center text-white shadow-sm transition-colors duration-300');
  });

  // Snapshot test
  it('matches snapshot', () => {
    const { container } = render(<Button {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});