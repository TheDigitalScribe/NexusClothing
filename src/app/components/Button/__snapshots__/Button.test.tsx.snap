// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<div>
  <button
    aria-label="Test Aria Label"
    class="rounded-full py-2 px-4 bg-blue-600 items-center text-white shadow-sm transition-colors duration-300"
  >
    Test Text
  </button>
</div>
`;
